lib/spack/llnl/util/filesystem.py

```diff
@@ -1190,20 +1199,15 @@ def remove_linked_tree(path):
     Parameters:
         path (str): Directory to be removed
     """
-    # On windows, cleaning a Git stage can be an issue
-    # as git leaves readonly files that Python handles
-    # poorly on Windows. Remove readonly status and try again
-    def onerror(func, path, exe_info):
-        os.chmod(path, stat.S_IWUSR)
-        try:
-            func(path)
-        except Exception as e:
-            tty.warn(e)
-            pass
+    # On windows, certain readonly files
+    # cannot be removed by Python
+    # Ensure proper permissions
+    if is_windows:
+        def make_removable(path):
+            os.chmod(path, stat.S_IWUSR)
```

Comment:
On linux the typical approach for modifying file permissions is

```
old_permissions = os.stat(the_file)
os.chmod(the_file, old_permissions | new_permission)
```

Does that not work on Windows?

Reply:
From the docs: 

> Note Although Windows supports [chmod()](https://docs.python.org/3/library/os.html#os.chmod), you can only set the file’s read-only flag with it (via the stat.S_IWRITE and stat.S_IREAD constants or a corresponding integer value). All other bits are ignored.